### PR TITLE
Azure ad auth

### PR DESCRIPTION
**Pull request by kaiyangzheng**
## Description/Problem

Closes [story](https://www.notion.so/sandboxnu/0ff8d90722bf435db9475c07e2311a2b?v=f7473b246d1742e395395cf955037121&p=f71afae0351f450e9722b842799244d4&pm=s)

Use Azure Entra ID to authenticate users

## Solution

Wrapped current auth with NextAuth to support Entra ID authentication.

## Dependencies

[x] This PR adds new dependencies

## Testing

Login with user and signout

### CODE REVIEW

File: apps/proxy.ts
Comment (AnshulShirude) [inline apps/proxy.ts:1]: What is the purpose of v1?

File: apps/server/src/app.controller.ts
Comment (AnshulShirude) [inline apps/server/src/app.controller.ts:1]: Good catch!

**Review by AnshulShirude — COMMENTED** [review]
Nice work!

File: apps/proxy.ts
Comment (kaiyangzheng) [reply apps/proxy.ts:1]: NextAuth uses /api/auth as their authentication endpoint, so I added v1 to all the backend endpoints to differentiate them in the proxy

**Review by AnshulShirude — APPROVED** [review]

**Review by ahweigl — APPROVED** [review]
🚀

**Pull request merged**